subsys/bluetooth/services/ots/ots.c

```diff
@@ -78,6 +78,55 @@ static ssize_t ots_obj_name_read(struct bt_conn *conn,
 				 strlen(ots->cur_obj->metadata.name));
 }
 
+#if CONFIG_BT_OTS_OBJ_NAME_WRITE_SUPPORT
+ssize_t ots_obj_name_write(struct bt_conn *conn,
+			   const struct bt_gatt_attr *attr,
+			   const void *buf, uint16_t len,
+			   uint16_t offset, uint8_t flags)
+{
+	struct bt_ots *ots = (struct bt_ots *) attr->user_data;
+	struct bt_gatt_ots_object *obj = NULL;
+	int rc = 0;
+	char name[CONFIG_BT_OTS_OBJ_MAX_NAME_LEN + 1];
+
+	LOG_DBG("OTS Object Name GATT Write Operation");
+
+	if (!ots->cur_obj) {
+		LOG_DBG("No Current Object selected in OTS!");
+		return BT_GATT_ERR(BT_GATT_OTS_OBJECT_NOT_SELECTED);
+	}
+
+	if (offset > 0) {
+		LOG_DBG("Rejecting a long write, offset must be 0!");
+		return BT_GATT_ERR(BT_GATT_OTS_WRITE_REQUEST_REJECTED);
+	}
+
+	if (offset + len > CONFIG_BT_OTS_OBJ_MAX_NAME_LEN) {
```

Comment:
Since we have the above check for `offset > 0`
```suggestion
	if (len > CONFIG_BT_OTS_OBJ_MAX_NAME_LEN) {
```